### PR TITLE
tests: Add simulation sweeps in CircleCI build.

### PR DESCRIPTION
**Pull request by yaoyuannnn**
We previously added the support in CircirCI for downloading the latest
gem5 binary from gem5-aladdin repo, but didn't actually use the
downloaded binary. This invokes the SMAUG sweep generator and runs the
generated sweeps using the downloaded gem5 binary.

We currently sweep the number of accelerators (1-4) and the SoC interface
choices (DMA or ACP). Because our current CircleCI plan only has 4GB
memory, we can only run one simulation at a time in order to not go OOM.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,22 +15,28 @@ jobs:
...
           command: |
             export PYTHONPATH=$SMAUG_HOME/build:$PYTHONPATH
             make test-run
+      - run:
+          name: Run simulation sweeps
+          command: |
+            # Download the gem5 binary.
+            python .circleci/download_artifacts.py --api_token=${GEM5_ALADDIN_BUILD_ARTIFACT_TOKEN} --project=gem5-aladdin --artifact_name=gem5.opt --user=${USER} --download_loc=/tmp --filter=${BUILD_ARTIFACT_FILTER} --branch=${BUILD_ARTIFACT_BRANCH}
+            chmod +x /tmp/gem5.opt
+            ln -s /usr/lib/x86_64-linux-gnu/libtcmalloc_minimal.so.4.3.0 /usr/lib/x86_64-linux-gnu/libtcmalloc.so.4
```
Comment (xyzsam) [inline .circleci/config.yml:31]: Hm, why is this necessary? I thought we use (basically) the same Docker image to build gem5-aladdin?

File: .circleci/run_simulation_sweeps.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import os
+import subprocess
+import six
+import shutil
+import tempfile
+
+models= ["minerva"]
+
+def run_sim_sweeps(sweep_file, gem5_binary, num_threads):
+  sweeps_dir = "%s/experiments/sweeps" % os.environ["SMAUG_HOME"]
```
Comment (xyzsam) [inline .circleci/run_simulation_sweeps.py:11]: os.path.join

File: .circleci/run_simulation_sweeps.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  sweep_generator = os.path.join(sweeps_dir, "main.py")
+
+  for model in models:
+    temp_dir = tempfile.mkdtemp(dir=os.getcwd())
+    process = subprocess.Popen([
+        "python", sweep_generator, "--model", model, "--params", sweep_file,
```
Comment (xyzsam) [inline .circleci/run_simulation_sweeps.py:17]: I was confused where this file went, then I realized it's in the experiments submodule. We should move as much of the code from that into here, so experiments is literally just a dumb directory full of model files, parameters, and simulation configuration files. I have not carefully looked at even a single line of code in that main.py file yet.

**Review by xyzsam — APPROVED** [review]
Forgot to approve.

File: .circleci/run_simulation_sweeps.py (hunk 3 above)
Comment (yaoyuannnn) [reply .circleci/run_simulation_sweeps.py:17]: Agreed. I will move the sweep generator to here.

File: .circleci/run_simulation_sweeps.py (hunk 2 above)
Comment (yaoyuannnn) [reply .circleci/run_simulation_sweeps.py:11]: Done.

File: .circleci/config.yml (hunk 1 above)
Comment (yaoyuannnn) [reply .circleci/config.yml]: Found the culprit. I didn't install `libgoogle-perftools-dev` in the SMAUG docker image, which caused the issue. Added a commit for that.
